atest/testdata/running/for.robot

```diff
@@ -317,6 +324,120 @@ For In Range With Expressions Containing Floats
+    @{items}=    Create List    a    b    c    d
+    @{things}=    Create List    e    f    g    h
+    @{stuffs}=    Create List    1    2    3    4    5
+    : FOR    ${item}    IN ZIP    ${items}    ${things}    ${stuffs}
+    \    Fail    This test case should die before running this.
+
+For In Zip With Too Many Variables
+    [Documentation]    FAIL    Expected 3 Loop variables, but found 4
+    @{items}=    Create List    a    b    c    d
+    @{things}=    Create List    e    f    g    h
+    @{stuffs}=    Create List    1    2    3    4    5
+    : FOR    ${item}    ${thing}    ${flotsam}    ${jetsam}    IN ZIP    ${items}    ${things}    ${stuffs}
+    \    Fail    This test case should die before running this.
+
+For In Enumerate (with 4 items)
+    @{items}=    Create List    a    b    c    d
+    : FOR    ${index}    ${item}    IN ENUMERATE    @{items}
+    \    Should Be Equal    @{items}[${index}]    ${item}    Loop value ${item} should be item ${index} of ${items}
+    \    @{result}=     Create List    @{result}    ${index}:${item}
+    Should Be True    @{result} == ['0:a', '1:b', '2:c', '3:d']
+
+For In Enumerate (with 5 items)
+    @{items}=    Create List    a    b    c    d    e
+    : FOR    ${index}    ${item}    IN ENUMERATE    @{items}
+    \    Should Be Equal    @{items}[${index}]    ${item}    Loop value ${item} should be item ${index} of ${items}
+    \    @{result}=     Create List    @{result}    ${index}:${item}
+    Should Be True    @{result} == ['0:a', '1:b', '2:c', '3:d', '4:e']
+
+For In Enumerate With 3 Variables
+    @{items}=    Create List    a    b    c    d    e    f
+    : FOR    ${index}    ${item}    ${another_item}    IN ENUMERATE    @{items}
+    \    Should Be Equal    @{items}[${index* 2 }]    ${item}    Loop value ${item} should be item ${index} of ${items}
+    \    Should Be Equal    @{items}[${index * 2 + 1}]    ${another_item}    Loop value ${another_item} should be item ${index} of ${items}
+    \    @{result}=     Create List    @{result}    ${index}:${item}:${another_item}
+    Should Be True    @{result} == ['0:a:b', '1:c:d', '2:e:f']
+
+For In Enumerate With not the right number of variables
+    [Documentation]    FAIL    Number of FOR loop values should be multiple of variables. Got 3 variables (2 of which matter for this error) but 7 values.
+    @{items}=    Create List    a    b    c    d    e    f    g
+    : FOR    ${index}    ${item}    ${another_item}    IN ENUMERATE    @{items}
+    \    Should Be Equal    @{items}[${index* 2 }]    ${item}    Loop value ${item} should be item ${index} of ${items}
+    \    Should Be Equal    @{items}[${index * 2 + 1}]    ${another_item}    Loop value ${another_item} should be item ${index} of ${items}
+    \    @{result}=     Create List    @{result}    ${index}:${item}:${another_item}
+    Should Be True    @{result} == ['0:a:b', '1:c:d', '2:e:f']
+
+For In Enumerate With Too Few Variables
+    [Tags]    Not Ready
```

Comment:
These tags won't do anything on this side. I also assume these tests are already ready.